Add redux aa1 to implement 1

Aichnerc opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently Redux is not implemented.


**What is the new behavior (if this is a feature change)?**
- Actions and reducers have been added and were successfully implemented.

Aichnerc committed:
Implement connector actions

The basic actions for the connectors have been implemented. Dummy data is present.

Aichnerc committed:
Implement gitlab actions

The basic actions for gitlabs have been implemented. Dummy data is present.

Aichnerc committed:
Improve gitlab action

A comment has been changed.

Aichnerc committed:
Improve connector actions

Some dispatches have been adjusted.

Aichnerc committed:
Implement permission actions

The action has been implemented with dummy data.

Aichnerc committed:
Implement pipelines actions

The action has been implemented with dummy data.

Aichnerc committed:
Implement connector reducer

The reducer has been implemented.

Aichnerc committed:
Implement permission reducer

The reducer has been implemented.

Aichnerc committed:
Implement pipeline reducer

The reducer has been implemented.

Aichnerc committed:
Improve gitlab actions

When creating a gitlab, the new gitlab is also passed to reducer.

Aichnerc committed:
Implement gitlab reducer

The reducer has been implemented.

Aichnerc committed:
Implement root reducer

The root reducer is combining all reducers and makes them accessible.

Aichnerc committed:
Implement redux

Redux can now be used in the application.

Aichnerc committed:
Remove gitkeep

The files are no longer needed, once there is now content.

Aichnerc committed:
Temporarily remove Routes

The routes have not been adjusted yet, so the application will crash if they are present.

schettn commented on src/store/actions/pipelineActions.js:
```diff
@@ -0,0 +1,48 @@
+      {
+        domain: "share.local",
+        org: "anexia",
+        title: "Hydra",
```
@Aichnerc Neh..

schettn reviewed (approved):
Looks good for now.
Amore intensive review will take place in the master merge.

Please remove all those stuff which SNEK doesn't stand for.. You know what I mean.

Aichnerc committed:
Remove fascist organisation

Hail Hydra though.

pinterid reviewed (approved):
Looks good.

schettn commented on src/store/actions/connectorActions.js:
```diff
@@ -0,0 +1,212 @@
+            domain: "terminus.local",
+          },
+        ],
+        settings: {
+          shared: {
+            projects: [{ title: "Projekt 1" }, { title: "Projekt 2" }],
+            users: [
+              { full_name: "Florian Schett" },
+              { full_name: "Nico Kleber" },
+            ],
+            companyData: {
+              legal_name: "Werbeagentur Christian Aichner",
+            },
+            statistics: {
+              memberOrigin: false,
+            },
+          },
+          contribData: {
+            commits: [{ total: 1 }],
+          },
+        },
+      },
+    ];
```
The settings have to take care of whether data such as projects, user, etc are displayed or not. The dummy data must also correspond to this.

schettn replied:
Adjust this in the whole file.

Aichnerc replied:
I'll adjust that on the go when implement actual dummy data functionality. I am aware of this, yes.

schettn commented on src/store/actions/connectorActions.js:
```diff
@@ -0,0 +1,212 @@
+
+// Get connector by id
+export const getConnectorById = (id) => {
+  return (dispatch, getState, { clientShopify }) => {
```
looks like shopify xD

schettn reviewed (changes requested):
There is some invalid stuff.. have a look.
Btw `clientShopify` is all over the code. I would change this xD

Aichnerc committed:
Improve actions

shopfyClient has been replaced with getIntel due to a request by @schettn. This was an error caused by copy-paste.

schettn reviewed (approved):
LGTM
